### DISCUSSION TITLE
How can I define a product with descriptions in more languages?

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore.Commerce#463] [Q&A] [answered]
How could I define a product containing the descriptions translated in English and other languages?
If I create an English product and an another one in Italian, they won't share the inventory.

**Comment by sarahelsaig** [comment, marked as answer]
If you enable the `OrchardCore.ContentLocalization` feature and the product contains `LocalizationPart`, you can add localized versions of the product.

![image](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/4007293/1cd82999-f10b-489e-ba82-fd2eb595d510)

Unfortunately simple `InventoryPart`-based inventory tracking is not supported for this scenario, because of the design limitations of OC's `OrchardCore.ContentLocalization` module. (Localized content items are clones at the time of creation with only a reference to each other, they can't share/synchronize content part data.) You have two options:
- First, replace `LocalInventoryProvider` with your own `IProductInventoryProvider` implementation that stores the inventory registry somewhere else independent of the content item content (e.g in a site setting, auxiliary database table, etc).
- Alternatively, if you only have to translate descriptions (i.e. if it's not a problem that the `ContentItem.DisplayText` won't be translatable), you can place the description in a `LiquidPart` and [use Liquid to translate the contents of the description](https://docs.orchardcore.net/en/latest/reference/modules/Liquid/#t) in-place like this: `{{ "The content of the description!" | t }}`.

**Reply by DotCat1985** [reply]
I created my own `IProductInventoryProvider`, but it seems not to work when I use this snippet on `Startup.cs` (I'm developing a project in **.NET 8**, **OrchardCore 1.8.3** and **OrchardCore.Commerce 2.1.0**):

```
services.Replace(
          ServiceDescriptor.Scoped<IProductInventoryProvider, MyServices.LocalInventoryProvider>()
);
```

What could you suggest to solve my issue?

**Reply by sarahelsaig** [reply]
Are you sure your module's startup is called _after_ `OrchardCore.Commerce.Startup.ConfigureServices`? Make sure `"OrchardCore.Commerce"` is added to the dependencies in your _Manifest.cs_.

Also you can check what services are registered in any of your own services or controllers by injecting `IEnumerable<IProductInventoryProvider>` in the constructor. (injecting `IEnumerable<IService>` instead of just `IService` gives you an array of all registered implementations)
